apps/docs/src/guide/contracts/optimising-contract-calls.md

```diff
@@ -0,0 +1,106 @@
+# Optimising Contract Calls
+
+When you call a contract function via the SDK, it makes two major network requests:
+
+1. **Preparing the transaction**: This involves creating the transaction request, fetching dependencies for it, and funding it properly.
+
+2. **Sending the transaction**: This involves sending the transaction to the network and waiting for it to be confirmed.
+
+The below flowchart shows this entire process:
+
+![Transaction Lifecycle in the SDK without prefetching](/public/txdep1.png)
+
+Since the SDK prepares the contract call _after_ the user presses 'Submit Transaction' (or your code invokes the `call` method), the chain feels slower to the user than it actually is. This is because of the network request needed to prepare the transaction.
+
+This can be mitigated by preparing the contract call _before_ the user presses 'Submit Transaction'. If the transaction is prepared beforehand, the SDK only has to send the transaction to the network and wait for it to be confirmed. This reflects the actual speed of the chain to the user.
```

Comment:
```suggestion
This can be mitigated by preparing the contract call _before_ the user submits the transaction. If the transaction is prepared beforehand, the SDK only has to send the transaction to the network and wait for it to be confirmed. This reflects the actual speed of the chain to the user.
```